### PR TITLE
feat(compute): add compute disk create secondary regional sample

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description
Documentation - https://cloud.google.com/compute/docs/disks/async-pd/configure?authuser=0#gcloud

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  // Creates a secondary disk in a specified region.
+  public static Status createDiskSecondaryRegional(String projectId,
+       String secondaryProjectId, String primaryDiskName, String secondaryDiskName,
+       String primaryDiskRegion, String secondaryDiskRegion, long diskSizeGb, String diskType)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // An iterable collection of zone names in which you want to keep
+    // the new disks' replicas. One of the replica zones of the clone must match
+    // the zone of the source disk.
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java:71]: I don't think this is mandatory. Kindly remove this comment and let the docs describe about the requirements.

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    List<String> replicaZones = Arrays.asList(
+        String.format("projects/%s/zones/%s-c", secondaryProjectId, secondaryDiskRegion),
+        String.format("projects/%s/zones/%s-b", secondaryProjectId, secondaryDiskRegion));
+
+    String primaryDiskSource = String.format("projects/%s/regions/%s/disks/%s",
+        projectId, primaryDiskRegion, primaryDiskName);
+
+    DiskAsyncReplication asyncReplication = DiskAsyncReplication.newBuilder()
+        .setDisk(primaryDiskSource)
+        .build();
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java:81]: Kindly move this code within the try block

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java (hunk 1 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java:71]: Thank you! Deleted

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java (hunk 2 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java:81]: Thank you! Fixed

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+          .setName(secondaryDiskName)
+          .setSizeGb(diskSizeGb)
+          .setType(diskType)
+          .setRegion(secondaryDiskRegion)
+          .setAsyncPrimaryDisk(asyncReplication)
+          .build();
+
+      // Wait for the create disk operation to complete.
+      Operation response = disksClient.insertAsync(secondaryProjectId, secondaryDiskRegion, disk)
+          .get(3, TimeUnit.MINUTES);
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java:98]: As for other operations, 3 minutes was enough.

[the remark this answers is not in the document]
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java:94]: Is it the same case with this operation as well? We don't want flaky tests.

File: compute/cloud-client/src/test/java/compute/disks/DisksIT.java [hunk 4]
```diff
@@ -317,6 +317,7 @@ public void testCreateReplicatedDisk()
             replicaZones, REGIONAL_REPLICATED_DISK, 100, DISK_TYPE);
 
     assertThat(status).isEqualTo(Status.DONE);
+    assertThat(status).isEqualTo(Status.DONE);
```
Comment (Sita04) [inline compute/cloud-client/src/test/java/compute/disks/DisksIT.java:320]: Why is this assert statement duplicated here?

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/test/java/compute/disks/DisksIT.java (hunk 4 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/disks/DisksIT.java:320]: It's a mistake. Thank you! Deleted

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java:94]: We also set it to 3 minutes for other disk creation operations. Why should it be different in this case?

[the remark this answers is not in the document]
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java:94]: Just confirming if this is enough time. Not opposing your claim.

**Review by Sita04 — APPROVED** [review]

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java:94]: There is enough time. The test passed without errors.

**Pull request merged**